src/game/TileEngine/WorldDef.cc

```diff
@@ -227,8 +227,8 @@ try
 			// ATE: If here, don't load default surface if already loaded
 			if (gbDefaultSurfaceUsed[i]) continue;
 
-			filename       = gTilesets[GENERIC_1].TileSurfaceFilenames[i];
-			tileset_to_add = GENERIC_1;
+			tileset_to_add = (tileset_id < DEFAULT_JA25_TILESET || i == SPECIALTILES) ? GENERIC_1 : DEFAULT_JA25_TILESET;
```

Comment:
Why is the extra SPECIALTILES check needed? If it is, why just here?

Reply:
It is ported from [PATCH119 (R440)](https://github.com/ja2-derek/ja2-ub-comparison/commit/a9d2a98a#diff-f80efc555e07bb1af95205bd9b05cbe8R440). 

I believe the `DEFAULT_JA25_TILESET` do not have `SPECIALTILES`. So we need to fallback one more step to `GENERIC_1`. Maybe it is clearer to negate it?

```c
 tileset_to_add = (tileset_id >= DEFAULT_JA25_TILESET && i != SPECIALTILES) ? DEFAULT_JA25_TILESET : GENERIC_1;
```

Reply:
I think it'd be clea[nr]est if it was outsourced to a function, since we do these lookups in three places.